kvstore/dbr/dbr.py

```diff
@@ -0,0 +1,85 @@
+import time
+from uuid import uuid4
+from pprint import pformat
+from asyncio import TaskGroup
+from kvstore.dbr.dbr_environment import DBREnvironment
+from kvstore.utils.enums import DBRStatus
+from kvstore.utils.globals import LOGGER, CLIENT_LOCATION
+
+
+# add static typing w/ mypy and pylint
+class DBR:
+    """
+    A DBR (DBRequest) is recursively defined as a chain of DBRs. A single DBR is the largest subset of parallelizable tasks in the program
+    """
+
+    def __init__(
+        self,
+        name=None,
+        predecessor=None,
+        successor=None,
+        environment=None,
+    ):
+        """
+
+        Initialize a DBRequest
+
+        """
+        self.id = uuid4()
+        self.name = name
+        self.status = DBRStatus.DBR_CREATED
+        self.dbts = {}  # mapping between query ID and dbt
+        self.predecessor = predecessor
+        self.successor = successor
+        self.environment = environment
+
+        # Default placement
+        self.location = CLIENT_LOCATION
+        if self.predecessor:
+            self.location = (
+                self.predecessor.location
+            )  # default scheduling next to predecessor, smart placement can change this
```

Comment:
imo, this should be set in one place - i.e. we shouldn't default it to a location at initialization time, it should just default to None (which is nice bc it'll throw an error if something is broken). 

alternatively, we can shift location-setting logic out to a function, which gets called through `DBR.execute`